File: Course 4 - Algorithms on Strings/assignment 1/trie_matching/trie_matching.py
# python3
import sys

NA = -1

class Node:
	def __init__ (self):
		self.next = [NA] * 4

def solve (text, n, patterns):
	result = []

	// write your code here

	return result

text = sys.stdin.readline ().strip ()
n = int (sys.stdin.readline ().strip ())
patterns = []
for i in range (n):
	patterns += [sys.stdin.readline ().strip ()]

ans = solve (text, n, patterns)

sys.stdout.write (' '.join (map (str, ans)) + '\n')

File: Course 4 - Algorithms on Strings/trie_matching/starters_by_learners/trie_matching.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace trie_matching
{
    class Program
    {
        static void Main(string[] args)
        {
            string text = Console.ReadLine();
            int n = int.Parse(Console.ReadLine());
            List<string> patterns = new List<string>();
            for (int i = 0; i < n; i++)
            {
                string s = Console.ReadLine();
                patterns.Add(s);
            }

            List<int> answers = Solve(text, patterns);
            string answersLine = string.Join(" ", answers);
            Console.WriteLine(answersLine);
        }

        static List<int> Solve(string text, List<string> patterns)
        {
            List<int> ans = new List<int>();
            //write your code here

            return ans;
        }
    }
}